django_password_validators/password_history/password_validation.py

```diff
@@ -42,26 +60,48 @@ def validate(self, password, user=None):
 
         if not self._user_ok(user):
             return
-
         for user_config in UserPasswordHistoryConfig.objects.filter(user=user):
             password_hash = user_config.make_password_hash(password)
             try:
-                PasswordHistory.objects.get(
-                    user_config=user_config,
-                    password=password_hash
-                )
-                raise ValidationError(
-                    _("You can not use a password that is already used in this application."),
-                    code='password_used'
-                )
+                if self.lookup_range == 0:
+                    raise PasswordHistory.DoesNotExist
+
+                current_user_passwords = PasswordHistory.objects.filter(
+                    user_config=user_config
+                ).order_by('date')
+
+                num_user_passwords = len(current_user_passwords)
+
+                if self.lookup_range >= num_user_passwords:
```

Comment:
@spokerman12, why is this `if` condition needed? Wouldn't this cause the password history not to be checked if it is false?

Reply:
> The number of passwords in the history should already be <= lookup_range

One of the edge cases.

During the execution of `validation`, `lookup_range` can be either larger, lesser or equal than `num_user_passwords` because the setting can be changed during downtime, and `validation` runs before `password_changed`.

If  `lookup_range >= num_user_passwords`, a `.get` is enough.
If `lookup_range < num_user_passwords`, we have to go over the lookup_range so we still need the `if any` we removed just now. I'll undo it.